lib/chef/node/attribute.rb

```diff
@@ -216,16 +216,12 @@ def initialize(normal, default, override, automatic, node = nil)
        # that precedence level, +value+ will be the symbol +:not_present+.
       def debug_value(*args)
         COMPONENTS.map do |component|
-          ivar = instance_variable_get(component)
-          value = args.inject(ivar) do |so_far, key|
-            if so_far == :not_present
-              :not_present
-            elsif so_far.has_key?(key)
```

Comment:
`Array#has_key?` right there was the bug -- the read/read! APIs already fixed this and have tests.